### DISCUSSION TITLE
The new release does not start on A95X F2(S905X2)

### DISCUSSION THREAD

**Discussion by yrksly** [devmfc/debian-on-amlogic#14] [Q&A]
The previous two versions worked and launched without problems, but the new one with kernel 6.6.1 does not start, just set LOAD and do not boot any further.

And another question - how to install this image into the internal memory, the installation works with the script, but when I try to start it only says “boot”

**Comment by devmfc** [comment]
Strange, I own a A95X F2 myself and it just boots fine. Do you boot via USB (I recommend this) or via SD card? And if you boot via SD card, did you change the boot.confg accordingly? Which box did you select in boot.config?

For the second question, which script did you use?

**Comment by yrksly** [comment]
> Strange, I own a A95X F2 myself and it just boots fine. Do you boot via USB (I recommend this) or via SD card? And if you boot via SD card, did you change the boot.confg accordingly? Which box did you select in boot.config?
> 
> For the second question, which script did you use?

I launch from an SD card, the box worked for me for a month as a nextcloud server, but after a restart it didn’t turn on anymore) I saw a new release, I decided to try to launch it from the internal memory, but it doesn’t even start from the SD card. Now I'll try to launch from USB.

I use this script - https://gist.github.com/devmfc/d6b3a5cfbb2802b61bea5b799a03498b

**Comment by yrksly** [comment]
If you start from USB and select A95X F2 in boot.conf, then everything works, but it doesn’t start via an SD card with the same config, but if you just select S905X2, then it starts from the SD card. Although everything worked in previous releases. And the issue with installation into internal memory was not resolved...

**Comment by devmfc** [comment]
I tried different combinations on my own A95X F2, but I can't reproduce any problem. Are you sure you have *root=/dev/mmcblk0p2* enabled in boot.config when you use an SD card?

That script file is very experimental. There is a lot of things that can break that script.... Did you run `./aml-multiboot-setup.sh` before the install script?

**Reply by yrksly** [reply]
The latest release could not be launched from the SD card, but the previous version was installed into the internal memory after using the multiboot script! Now I'll try to install the latest release from USB.

upd: I installed the latest release via USB, configured the nextcloud and transmission and will continue testing) Thanks for the help and the second life of the TVbox after the HDMI on it failed!)))

**Reply by devmfc** [reply]
Great to hear!  Have fun.

**Comment by Lelik-74** [comment]
hello, I have a95x f2 4GB+32GB Wifi 2.4G - AMPAK AP6212, you can make a dtb file for it. Wifi and bluetooth does not work. Thanks!

**Comment by dreamlayers** [comment]
I have an OTT TV Box, model ABOX-A1 ( 1G + 8G, S905X ). Inside it looks very similar to NEXBOX A95X discussed at https://www.savagemessiahzine.com/forum/index.php?showtopic=770346

I could not boot Devmfc_Debian-Bookworm_6.6.1-meson64_Minimal-23.11.09.img.xz but I could boot Devmfc_Debian-Bookworm_6.1.57-meson64_Minimal-23.10.12.img.xz. I tried both a microSDHC card and a USB drive. The failing one hung at the initial ABOX logo every time, and the working one boots to the login prompt. 

I simply did `xzcat file.xz | | sudo sh -c "cat > /dev/sdc"` to write the image, and then edited boot.config in the BOOT partition. There I uncommented `box=s905x_generic` and changed root on the SD card. I used a Q-tip with the cotton removed from one end to press the switch in the AV port.

**Reply by devmfc** [reply]
Did you try both images with the same SD card? 
Do you have a Serial  to USB adapter to see what is going while booting?
You can try to update the kernel of the working image with the newer kernel package of the newer image to see if that works? `dpkg -i linux-image-meson64_6.6.1~20231109_arm64.deb`

**Reply by dreamlayers** [reply]
Earlier, I was using the same microSDHC card, same card reader, and same USB drive for testing both versions. Then I decided to use Armbian instead, and that is currently on the USB drive. But I could play with the microSDHC card, and I just did some simple tests.

I copied the 6.1.57 kernel to the 6.6.1 boot partition, and that hung the same way, on the initial ABOX logo. Then I copied the 6.6.1 kernel and System.map to the 6.1.57 boot partition, and I got a screen of Linux kernel output text. I could never get any kernel output with the 6.6.1 image. So it seems the kernel is not the problem here, but something else causes the boot to fail. (I was overwriting the existing kernels in the boot partition, using their names, because that's what the other files look for.)

**Reply by dreamlayers** [reply]
The device tree seems to be at fault. I moved the dtb-6.6.1-meson64/amlogic folder to a different name, and then copied dtb-6.1.57-meson64/amlogic to that location. After that I booted to the login prompt using Devmfc_Debian-Bookworm_6.6.1-meson64_Minimal-23.11.09.img.xz with no other modifications.

**Reply by devmfc** [reply]
Thanks for troubleshooting that. 
I did not make any changes to my parts of the DTB and the upstream changes for the latest release have nothing really strange in it...  And I can't reproduce it on my s905x boxes...
So to really know what is going on you'll need a serial adapter to see what is going on during boot. Or you can do a dtdiff of the two DTB's and change the working DTB step by step until it fails. Than you know what part of the upstream changes breaks for your box.

**Reply by dreamlayers** [reply]
I gave this another try, and I think I found the problem. The following part of `dtb-6.6.1-meson64/amlogic/meson-gxl-p212-generic.dtb` causes a hang at the initial ABOX logo, with no kernel text output to HDMI. With this removed, I ran many tests showing kernel text output to HDMI, and a few booting all the way to the login prompt. All these tests were done using `Devmfc_Debian-Bookworm_6.6.1-meson64_Minimal-23.11.09.img.xz` and `box=s905x_generic` on my ABOX-A1 1 GB + 8 GB S905X.
```
		linux,secmon {
			alignment = <0x00 0x400000>;
			alloc-ranges = <0x00 0x5000000 0x00 0x400000>;
			compatible = "shared-dma-pool";
			reusable;
			size = <0x00 0x400000>;
		};
````

BTW. Where does `meson-gxl-p212-generic.dtb` come from? In `box-config/box-s905x_generic.config` I see `dtb_img=dtb-${release}/amlogic/meson-gxl-p212-generic.dtb` and that is the dtb file I've been altering. But a Google search for "meson-gxl-p212-generic" does not find results. This does not seem to be a part of Linux kernel releases. Instead they build `meson-gxl-s905x-p212.dtb`. I thought that would be used for s905x_generic. That file exists in the boot partition, and differs from `meson-gxl-p212-generic.dtb`.

**Reply by devmfc** [reply]
That dtb is my patched version of meson-gxl-s905x-p212.dtb for android tv boxes. I will take a look at that reserved-memory node you found. It looks like something from an older kernel that somehow has slipped into my patched dtb... I didn't notice that before because it is in a dtsi I use.  Apparently on your box that is causing troubles. On my boxes it managed to stay unnoticed. Sorry...

**Reply by devmfc** [reply]
You can BTW also try to use meson-gxl-s905x-p212.dtb for your box, I think they don't differ that much anymore.

**Reply by dreamlayers** [reply]
Thank you. Devmfc_Debian-Bookworm_6.6.8-meson64_Minimal-23.12.22.img.xz works for me on my ABOX-A1 with `box=s905x_generic`.

`dtdiff` shows many differences between `meson-gxl-s905x-p212.dtb` and `meson-gxl-p212-generic.dtb`. This may mostly be changes of the phandle identifiers for things, and then changes where other things refer to those. I have not analysed this in depth. One change that does make a difference is Bluetooth being present in `meson-gxl-s905x-p212.dtb` but not in `meson-gxl-p212-generic.dtb`. That Bluetooth hardware is not present in my ABOX-A1, and booting with `meson-gxl-s905x-p212.dtb` results in some error messages, but no real problems.
